lib/src/linting/rules/no-objects-in-query-params.spec.ts

```diff
@@ -0,0 +1,162 @@
+  test("valid for correct usage", () => {
+    const errors = noObjectsInQueryParams({
+      api: fakeLocatable<ApiNode>({
+        name: fakeLocatable("example-api")
+      }),
+      endpoints: [
+        // Endpoint with inline query parameters
+        fakeLocatable<EndpointNode>({
+          name: fakeLocatable("listUsers"),
+          method: fakeLocatable<HttpMethod>("GET"),
+          path: fakeLocatable("/users"),
+          isDraft: false,
+          tests: [],
+          request: fakeLocatable<RequestNode>({
+            queryParams: fakeLocatable([
+              fakeLocatable<QueryParamNode>({
+                name: fakeLocatable("somequeryparam"),
+                type: {
+                  kind: TypeKind.STRING
+                },
+                optional: true
+              })
+            ])
+          }),
+          responses: []
+        }),
+        // Endpoint with reference type in query parameters
+        fakeLocatable<EndpointNode>({
+          name: fakeLocatable("createUser"),
+          method: fakeLocatable<HttpMethod>("POST"),
+          path: fakeLocatable("/users"),
+          isDraft: false,
+          tests: [],
+          request: fakeLocatable<RequestNode>({
+            queryParams: fakeLocatable([
+              fakeLocatable<QueryParamNode>({
+                name: fakeLocatable("somequeryparam"),
+                type: referenceType("somequeryparam", "", TypeKind.UNION),
+                optional: true
+              })
+            ])
+          }),
+          responses: []
+        })
+      ],
+      types: [
+        {
+          name: "somequeryparam",
+          type: unionType([STRING, INT32])
+        }
+      ]
+    });
+    expect(errors).toEqual([]);
+  });
+
+  test("rejects query parameters of type object", () => {
+    const errors = noObjectsInQueryParams({
+      api: fakeLocatable<ApiNode>({
+        name: fakeLocatable("example-api")
+      }),
+      endpoints: [
+        // Endpoint with request payload
+        fakeLocatable<EndpointNode>({
+          name: fakeLocatable("listUsers"),
+          method: fakeLocatable<HttpMethod>("GET"),
+          path: fakeLocatable("/users"),
+          isDraft: false,
+          tests: [],
+          request: fakeLocatable<RequestNode>({
+            queryParams: fakeLocatable([
+              fakeLocatable<QueryParamNode>({
+                name: fakeLocatable("somequeryparam"),
+                type: objectType([
+                  {
+                    name: "id",
+                    type: unionType([INT32, NULL]),
+                    optional: false
+                  }
+                ]),
+                optional: true
+              })
+            ])
+          }),
+          responses: []
+        })
+      ],
+      types: []
+    });
+    expect(errors).toEqual([
+      {
+        message:
+          "The type `somequeryparam` is of type object. Objects are not recommended in query parameters."
```

Comment:
Could we make the error message give more context, e.g. which endpoint this belongs to and what the name of the query parameter is, so it's easier to fix the issue? Otherwise, this could be hard to address when you have a large contract with many endpoints.